test-data/unit/check-generics.test

```diff
@@ -996,7 +996,7 @@ IntNode[int](1, 1)
 IntNode[int](1, 'a')  # E: Argument 2 to "Node" has incompatible type "str"; expected "int"
 
 SameNode = Node[T, T]
-ff = SameNode[T](1, 1)  # E: Need type annotation for 'ff'
+ff = SameNode[T](1, 1)
```

Comment:
I think that this should generate an error, since `T` is not in scope here. If this seems unrelated, you can create an issue about this (and add a TODO comment here).